airflow/providers/amazon/aws/transfers/mongo_to_s3.py

```diff
@@ -117,7 +117,6 @@ def execute(self, context) -> bool:
                 mongo_collection=self.mongo_collection,
                 query=cast(dict, self.mongo_query),
                 mongo_db=self.mongo_db,
-                allowDiskUse=self.allow_disk_use,
```

Comment:
hmm, this is an interesting one. The [`.find`](https://pymongo.readthedocs.io/en/stable/api/pymongo/collection.html?highlight=allowDiskUse#pymongo.collection.Collection.find) method seems to supports the argument but in snake case i.e. `allow_disk_use`. 
I'm not familiar with Mongo. Can you double-check if `allow_disk_use` is the correct one (it seems to be)?

Reply:
It's added in MongoDB version 4.4 for the `.find` method

Reply:
The `allow_disk_use` argument in `.find()` maps to MongoDB’s [`cursor.allowDiskUse`](https://docs.mongodb.com/manual/reference/method/cursor.allowDiskUse/), while `.aggregate()`’s `allowDiskUse` corresponds to [`allowDiskUse` in the aggregation pipeline](https://docs.mongodb.com/manual/reference/command/aggregate/#mongodb-dbcommand-dbcmd.aggregate). I’m honestly not familiar with `cursor.allowDiskUse` (in fact I didn’t know it existed until today), but from the documentation the two are quite different.

I think whether we should set `find(allow_disk_use=True)` depends on what we want `MongoToS3Operator.allow_disk_use` to mean. The docstring says

> allow_disk_use: in the case you are retrieving a lot of data, you may have to use the disk to save it instead of saving all in the RAM

which seems to indicate it probably makes sense to set `find(allow_disk_use=True)` from it. But then the question becomes how we can pass it only to MongoDB (not pymongo!) 4.4+ (released in July 2020) because it would crash on earlier versions.

Reply:
> I think whether we should set find(allow_disk_use=True) depends on what we want MongoToS3Operator.allow_disk_use to mean. 

Though the wording seems quite different, they seem to achieve the same thing: _use temporary files to store data when it exceeds certain memory limits (both 100 MB RAM)_ - one is for aggregation stage while the other is for blocking sort operation. But again, I'm not familiar with Mongo enough to be confident in what I said. It will be nice if someone already using Mongo can verify this.

> But then the question becomes how we can pass it only to MongoDB (not pymongo!) 4.4+ (released in July 2020) because it would crash on earlier versions.

I think a note in the docstring saying that it requires MongoDB 4.4+ and a link to the doc should be enough. It's not sensible for us to control the version of an external system. IMO, the users should know and control what MongoDB version they are using.

Reply:
Not ideal, but it seems to be the only way:
```python
find_method = partial(MongoHook(self.mongo_conn_id).find, 
                mongo_collection=self.mongo_collection,
                query=cast(dict, self.mongo_query),
                mongo_db=self.mongo_db)

if self.allow_disk_use:
    results = find_method(allow_disk_use=self.allow_disk_use)
else:
    results = find_method()

Reply:
@amatellanes  I'm ok with just removes it for now, and we can think about how to add it back in another PR in the future when someone requests it. Or we can handle the branching logic in this PR so that _allow_disk_use_ can play with single query as well.

If you wanna go with the former, please update the docstring to reflect the fact that _allow_disk_use_ doesn't apply to single query (`.find`).

Reply:
> I think a note in the docstring saying that it requires MongoDB 4.4+ for `.find` and a link to the doc should be enough. It's neither sensible for us to control the version of an external system nor that we can. IMO, the users should know and control what MongoDB version they are using.

The issue is `aggregate(allowDiskUse=True)` does work prior to MongoDB 4.4, and users running older MongoDB setups may still want to use that. If we always also pass the argument to `find()`, they will have to create two `MongoHook` instances, one with `allow_disk_use` and one without, which doesn’t feel like good interface design to me.

Reply:
Why two `MongoHook` instances? I think our discussion only applies to the `.find`, and the `.aggregate` will stay as it's.

Reply:
Ah, I misread and thought `allow_disk_use` were a property on `MongoHook` (it’s on `MongoToS3Operator` instead). Nvermind, I think your proposed solution above would work.

Reply:
@xinbinhuang I've just updated the docstring as you suggested.